sotodlib/core/axisman_io.py

```diff
@@ -85,6 +85,19 @@ def _retype_for_read(data):
 
 # save/load AxisManager to HDF5.
 
+def _safe_scalars(x):
+    # Kill any np.integer, np.floating, or np.str_ vals... they don't serialize.
+    if isinstance(x, list):
+        return [_safe_scalars(_x) for _x in x]
+    if isinstance(x, tuple):
+        return tuple([_safe_scalars(_x) for _x in x])
+    if isinstance(x, dict):
+        return {k: _safe_scalars(v) for k, v in x.items()}
+    if isinstance(x, (np.integer, np.floating, np.str_)):
```

Comment:
Probably an edge case since off the top of my head I can't think of a use case for it but things that are `np.complex` and `np.bool_` won't get caught by this. May be worth adding them just to cover the full gamut.

You could also just change the condition to `hasattr(x, "item")` which should catch everything.

Reply:
Ah, I will add np.bool_, thanks.

For complex values, I think it's ok if they hit the JSON converter and an error is raised.  We'd need a custom packer/unpacker.  Nothing in schema should be complex (although currently you can do core.LabelAxis('nth_roots', [(-1+0j)**(1/n)**i for i in range(n)]) ... but please don't ...)

Reply:
Yeah, LabelAxis with complex labels is probably a bad enough idea that it throwing an error seems preferable.